### ISSUE TITLE
Should be using [\t ]* instead of \s* in typeRegex

### ISSUE THREAD

**Issue by tougher** [Evertras/typescript-cs-poco#19]
Thanks for a nice gulp package!

If using `\s*` in the first match this will fail:

```
namespace SomeNamespace
{

    public enum JustAnEnum
    {
        One, Two
    }
}
```

It fails because the the `\1` backreference will include the extra new line.

This works for me:
`
var typeRegex = /^([\t ]*)(?:public\s*|partial\s*)*\s*(class|enum|struct)\s+([\w\d_<>]+)(?:\s*:\s*([\w\d\._]+))?\s*\{((?:.|\n|\r)*?)^\1\}/gm;
`

**Comment by Evertras** [comment]
I'm writing up a test with this and it seems to be handling it okay.  Still investigating.

**Comment by Evertras** [comment]
Sorry for taking so long on this.  I wasn't able to reproduce the problem.  However, your version is more correct for the intent anyway, so I made the change to your version in hopes it fixes your problem!  1.6.0 has the change.
